clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslClientAuthenticator.java

```diff
@@ -106,13 +107,21 @@ public void configure(TransportLayer transportLayer, PrincipalBuilder principalB
 
             setSaslState(handshakeRequestEnable ? SaslState.SEND_HANDSHAKE_REQUEST : SaslState.INITIAL);
 
-            // determine client principal from subject.
-            if (!subject.getPrincipals().isEmpty()) {
-                Principal clientPrincipal = subject.getPrincipals().iterator().next();
-                this.clientPrincipalName = clientPrincipal.getName();
-            } else {
-                clientPrincipalName = null;
+            // determine client principal from subject for Kerberos. For other mechanisms, the authenticated
+            // principal (username for PLAIN and SCRAM) is used as authorization id.
+            if (mechanism.equals(SaslConfigs.GSSAPI_MECHANISM)) {
+                try {
+                    Principal clientPrincipal = subject.getPrincipals().iterator().next();
```

Comment:
It seems a bit odd to invoke `next()` blindly. Can any of these calls throw an exception is `getPrincipals` is not empty?